### DISCUSSION TITLE
Autocompletion in IDE

### DISCUSSION THREAD

**Discussion by yarosdev** [laravel/framework#44065] [General]
Hello everyone, its 2022 ! And Laravel is the most known PHP framework in the world! 
So, why I am getting error message from my newest PhpStorm that "Method 'where' not found in \App\Models\Foo"?
Anyway, I have done following:

- sail artisan ide-helper:models
- plugin: Laravel Idea installed
- one more step?

It is the ugliest thing in the framework!

**Comment by jorqensen** [comment]
🤡

**Comment by henzeb** [comment]
You can help the developers with better typehinting. There are PR's concerning this so now and then...

**Comment by adelf** [comment]
Have you tried this? 

<img width="339" alt="Screenshot 2022-09-10 at 10 27 57" src="https://user-images.githubusercontent.com/2818394/189472123-dd3c209b-0024-464c-a9c2-28df094df0f0.png">

Also, the Foo class might not extend the Eloquent Model class...

**Comment by sethsandaru** [comment]
I'm using the free trial version & works super great. Will definitely subscribe https://laravel-idea.com

**Comment by Sophist-UK** [comment]
My project is open source but is so new it does not yet qualify! A real pity.
